### DISCUSSION TITLE
Can I perform computations on the vectors?

### DISCUSSION THREAD

**Discussion by ghost** [milvus-io/milvus#17345] [Q&A and General discussion]
For example, is Elasticsearch there's the Painless scripting language or custom plugins that using them I can perform custom algorithms on the data.
Is it possible with Milvus?

**Comment by yhmo** [comment]
Sounds like UDF(user define functions). No, so far milvus has no this ability.
Custom algorithms on vectors may have different purposes, such as dimensionality reduction, normalize, score by weight, custom metric type. What kind of algorithm are you suppose to apply to vectors?

**Reply by ghost** [reply]
@yhmo There are currently two purposes in our interest:
* Dimensionality reduction - For example, we want to perform T-SNE for a bucket of 100 million vectors
* Custom metric type - For example, given a distribution of vectors, 100 million vectors, we want to calculate the mean vector of this distribution
